src/App.tsx

```diff
@@ -4,32 +4,71 @@ import logo from "../public/FischLogo.svg";
 
 import styles from "./App.module.css";
 import { Card } from "./components/Card";
-import { useRef, useState } from "react";
+import { JSX, useEffect, useState } from "react";
+import { NewCard } from "./components/NewCard";
 
-function App() {
-  const [cardsAmount, setCardsAmount] = useState(1)
-  const [cards, setCards] = useState([{}]);
+interface AppProps {
+  basicCards: {
+    face: string;
+    back: string;
+  }[];
+}
+
+interface CardObject {
+  face: string;
+  back: string;
+}
+
+function App(props: AppProps) {
+  const importedCards = props.basicCards.map((card, index) => {
+    return { face: card.face, back: card.back, key: index };
+  });
+
+  const [cardsArray, setCardsArray] = useState(importedCards);
+
+  const [incrementalKey, setIncrementalKey] = useState(cardsArray.length);
+
+  useEffect(
+    () => setIncrementalKey((prevKey) => prevKey + 1),
+    [cardsArray, setCardsArray]
+  );
+
+  const [isNewCardDisplayed, setIsNewCardDisplayed] = useState(false);
 
   const handleAddCardButtonClick = () => {
-    setCardsAmount(cardsAmount+1)
-    setCards(prevCards => [...prevCards, {}]);
+    setIsNewCardDisplayed(true);
+  };
+
+  const handleSaveButtonClick = (props: CardObject) => {
+    setCardsArray([
+      { face: props.face, back: props.back, key: incrementalKey },
```

Comment:
**nitpick:** It looks good but we need to remove the `key` from here. It should not be set like this